opentaskspal/src/main/java/org/dmfs/opentaskspal/readdata/TaskDateTime.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2017 dmfs GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dmfs.opentaskspal.readdata;
+
+import android.support.annotation.NonNull;
+
+import org.dmfs.android.contentpal.Projection;
+import org.dmfs.android.contentpal.RowDataSnapshot;
+import org.dmfs.android.contentpal.projections.MultiProjection;
+import org.dmfs.jems.single.Single;
+import org.dmfs.rfc5545.DateTime;
+import org.dmfs.tasks.contract.TaskContract.Tasks;
+
+import java.util.TimeZone;
+
+
+/**
+ * {@link Single} for creating a {@link DateTime} from the date-time related properties of a task.
+ * Takes an already present timestamp, plus the separately stored timezone and all-day flag.
+ *
+ * @author Gabor Keszthelyi
+ */
+public final class TaskDateTime implements Single<DateTime>
+{
+    public static final Projection<Tasks> PROJECTION = new MultiProjection<>(Tasks.TZ, Tasks.IS_ALLDAY);
+
+    private final Long mTimestamp;
+    private final String mTimeZoneId;
+    private final boolean mIsAllDay;
+
+
+    public TaskDateTime(@NonNull Long timestamp, @NonNull String timeZoneId, @NonNull Boolean isAllDay)
```

Comment:
I completely disagree with this class. `TaskDateTime` should be an optional and take a `RowDataSnaphot` and a column name. The current implementation is not task related at all. It just constructs a DateTime from its components. A `TaskDateTime` which deserves the name should act as an adapter from task data to `DateTime`. Since most `DateTime` values of a task are optional this class should implement an `Optional` as well. See https://github.com/dmfs/opentasks/blob/stories/484-taskspal-subtask-data-alt/opentaskspal/src/main/java/org/dmfs/opentaskspal/readdata/TaskDateTime.java